### PR TITLE
#79 implement department overview

### LINKED ISSUE #79: Client: Implement page to view the different 'Ressorts'
A page which displays the different departments.

For each departments:
- Show the head of department
- Show the members of the department
- Show buttons for "Zur Organisation", "Zu den Zielen", "Zum Wiki-Artikel"

### PR DESCRIPTION
**Pull request by KevinGus**
Closes #79 
Implemented department overview page.

### CODE REVIEW

File: client/src/ts/members/DepartmentOverview.tsx [hunk 1]
```diff
@@ -0,0 +1,378 @@
+/**
+ * The RessortOverview-Component displays all members of a ressort/department and the actual leaders in a grid.
+ */
+import React, { useState, useEffect } from "react";
+import {
+  Button,
+  Paper,
+  Table,
```
Comment (johannesfbraun) [inline client/src/ts/members/DepartmentOverview.tsx:8]: Table - IconButton not used, so please remove

File: client/src/ts/members/DepartmentOverview.tsx [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+  IconButton,
+  Grid,
+  createStyles,
+  Theme,
+  makeStyles,
+} from "@material-ui/core";
+import PageBar from "../global/navigation/PageBar";
+import api from "../utils/api";
+import { NavLink } from "react-router-dom";
+import { NoEncryptionTwoTone } from "@material-ui/icons";
```
Comment (johannesfbraun) [inline client/src/ts/members/DepartmentOverview.tsx:25]: Not used please remove

File: client/src/ts/members/DepartmentOverview.tsx [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+        setErrorOpen(errorOpen + 1);
+      });
+
+    // Clean-up function
+    return () => {
+      mounted = false;
+    };
+  };
+
+  useEffect(() => {
```
Comment (johannesfbraun) [inline client/src/ts/members/DepartmentOverview.tsx:204]: @typescript-eslint/no-unused-vars
  Line 208:6:  React Hook useEffect has missing dependencies: 'getCurrentDirectors', 'getDepartmentMembers', and 'getDepartments'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

File: client/src/ts/members/DepartmentOverview.tsx [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+  const classes = useStyles();
+  const [members, setMembers] = useState<Member[]>([]);
+  const [departments, setDepartments] = useState<Department[]>([]);
+  const [directors, setCurrentDirectors] = useState<Director[]>([]);
+  const [errorOpen, setErrorOpen] = useState<number>(0);
+
+  /**
+   * Retrieves the departments
+   */
+  const getDepartments: VoidFunction = () => {
```
Comment (johannesfbraun) [inline client/src/ts/members/DepartmentOverview.tsx:126]: Gets a 304 Error

File: client/src/ts/members/DepartmentOverview.tsx [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+    // Clean-up function
+    return () => {
+      mounted = false;
+    };
+  };
+
+  /**
+   * Retrieves the department members
+   */
+  const getDepartmentMembers: VoidFunction = () => {
```
Comment (johannesfbraun) [inline client/src/ts/members/DepartmentOverview.tsx:153]: Gets a 404 error

File: client/src/ts/members/DepartmentOverview.tsx [hunk 6]
```diff
@@ -0,0 +1,378 @@
...
+    // Clean-up function
+    return () => {
+      mounted = false;
+    };
+  };
+
+  /**
+   * Retrieves the directors of the departments
+   */
+  const getCurrentDirectors: VoidFunction = () => {
```
Comment (johannesfbraun) [inline client/src/ts/members/DepartmentOverview.tsx:180]: Gets a 404 error

File: client/src/ts/members/DepartmentOverview.tsx [hunk 7]
```diff
@@ -0,0 +1,378 @@
...
+          return director.ressortID === 8;
+        });
+      }
+      default: {
+        return [];
+      }
+    }
+  };
+
+  return (
```
Comment (johannesfbraun) [inline client/src/ts/members/DepartmentOverview.tsx:304]: No data is displayed, as all the .get are returned an error message

**Review by johannesfbraun — CHANGES_REQUESTED** [review]
Please fix the errors with the .get methods

File: client/src/ts/members/DepartmentOverview.tsx [hunk 8]
```diff
@@ -0,0 +1,378 @@
...
+    return () => {
+      mounted = false;
+    };
+  };
+
+  useEffect(() => {
+    getDepartmentMembers();
+    getDepartments();
+    getCurrentDirectors();
+  }, []);
```
Comment (LennartRukower) [inline client/src/ts/members/DepartmentOverview.tsx:208]: This useEffect has missing dependencies.
(Follow the instructions of TSLint, which should already recommend dependencies).
See Issue #137 for a similiar use case.

File: client/src/ts/members/DepartmentOverview.tsx [hunk 9]
```diff
@@ -0,0 +1,378 @@
...
+        });
+      }
+      default: {
+        return [];
+      }
+    }
+  };
+
+  /**
+   * Filters cirectors in their departments
```
Comment (LennartRukower) [inline client/src/ts/members/DepartmentOverview.tsx:258]: Spelling mistake: "cirectors"

**Review by LennartRukower — CHANGES_REQUESTED** [review]

**Review by JanMueller31 — APPROVED** [review]
lgtm

**Pull request merged**